### PR TITLE
Traffic Manager version 2017-05-01 including new MonitorConfig settings

### PR DESCRIPTION
**Pull request by vladca**
## Description
Traffic Manager version 2017-05-01 including new MonitorConfig settings
https://github.com/Azure/azure-rest-api-specs/pull/1180
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/TrafficManager/Management.TrafficManager/generate.cmd [hunk 1]
```diff
@@ -4,7 +4,7 @@
 ::
 
 @echo off
-set autoRestVersion=0.16.0
+set autoRestVersion=0.13.2
```
Comment (shahabhijeet) [inline src/SDKs/TrafficManager/Management.TrafficManager/generate.cmd]: @vladca can you use the latest or at least 1.0 version of AutoRest.
Also can you use the commit Id as part of the argument in generate.cmd

File: src/SDKs/TrafficManager/Management.TrafficManager/project.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "2.2.0-preview",
+  "version": "2.3.0",
```
Comment (shahabhijeet) [inline src/SDKs/TrafficManager/Management.TrafficManager/project.json]: @vladca all traces of project.json, global.json and *.xproj should be deleted. These files are no needed anymore in the new project system.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
You will also need version change in csproj and assemblyInfo.cs
Also please start using latest ResourceManager to record your tests.
We will be removing references for ResourceManager from common dependencies.

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

File: src/SDKs/TrafficManager/Management.TrafficManager/generate.cmd (hunk 1 above)
Comment (vladca) [inline src/SDKs/TrafficManager/Management.TrafficManager/generate.cmd]: version: 0.13.2 is the latest, as per https://www.npmjs.com/package/autorest; I installed it as per instructions here https://github.com/Azure/autorest, using npm install -g autorest; it's the versioning that's all over the place (https://www.nuget.org/packages/AutoRest), but I have used the latest version; 

commit: I am waiting for @alvadb to merge https://github.com/Azure/azure-rest-api-specs/pull/1180

**Comment by vladca** [comment]
@shahabhijeet : "Also please start using latest ResourceManager to record your tests." : our test project is using version 1.1.0-preview of Microsoft.Azure.ResourceManager. I cannot find a newer version. 

"We will be removing references for ResourceManager from common dependencies." : what is the action for me here? when you remove said reference you will make sure all the projects are still working, right?

**Comment by shahabhijeet** [comment]
@vladca new versions are on nuget, you need to add that reference in your test projects.
And record all the tests that you have in your PR with new resource manager or if needed use new storage SDK.
Regarding removing existing dependency, yes the plan is to remove them and also make sure nothing is broken when we remove it.

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

File: src/SDKs/TrafficManager/TrafficManager.Tests/ScenarioTests/ProfileScenarioTests.cs [hunk 3]
```diff
@@ -12,7 +12,7 @@ namespace Microsoft.Azure.Management.TrafficManager.Testing.ScenarioTests
     using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
     using Xunit;
 
-    public class ProfileScenarioTests : TestBase
+    public partial class ProfileScenarioTests : TestBase
```
Comment (shahabhijeet) [inline src/SDKs/TrafficManager/TrafficManager.Tests/ScenarioTests/ProfileScenarioTests.cs]: @vladca this.CreateResourceGroup does not use the latest ResourceManager.
Even though you have added reference to the new nuget, it's not getting used.
You will have to either use it in this file or change the way ResourceManagement client is created in ResourceGroupHelper.cs

**Review by shahabhijeet — DISMISSED** [review]
Please re record your tests using new Resource Manager.
You have two choiced, if you can re record all that is the best option or at least start by recording tests that are part of this PR.

File: src/SDKs/TrafficManager/TrafficManager.Tests/ScenarioTests/ProfileScenarioTests.cs (hunk 3 above)
Comment (vladca) [inline src/SDKs/TrafficManager/TrafficManager.Tests/ScenarioTests/ProfileScenarioTests.cs]: That's done now, cheers

**Review by markcowl — APPROVED** [review]

**Pull request merged**